### DISCUSSION TITLE
Difference between extensions and distro packaging

### DISCUSSION THREAD

**Discussion by alvinhom** [open-telemetry/opentelemetry-java-instrumentation#6041] [Q&A]
I am writing a custom instrumentation plugin for our company.  I want to put it into a separate jar so I can get the latest opentelemetry jar and just build my custom plugin code.

I am reading the documentation in the examples directory and it looks like I should package it in extensions.  However, the distro package looks similar.  I am not quite clear on the differences.  Can someone explain the difference between them?

**Comment by trask** [comment]
hi @alvinhom! this is a good question. I have started an issue to document the differences: https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/6047.

In a nutshell, we recommend using the extension mechanism if that works for you as it is generally much simpler.
